srcalc/src/test/java/gov/va/med/srcalc/vista/vistalink/VistaLinkProcedureCallerTest.java

```diff
@@ -0,0 +1,67 @@
+package gov.va.med.srcalc.vista.vistalink;
+
+import static org.junit.Assert.*;
+import static org.mockito.Mockito.*;
+
+import java.util.Arrays;
+import java.util.List;
+
+import javax.naming.NamingException;
+import javax.resource.ResourceException;
+
+import gov.va.med.srcalc.vista.vistalink.VistaLinkProcedureCaller;
+import gov.va.med.vistalink.adapter.cci.VistaLinkConnectionFactory;
+import gov.va.med.vistalink.adapter.cci.VistaLinkConnectionSpec;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.springframework.mock.jndi.SimpleNamingContextBuilder;
+
+public class VistaLinkProcedureCallerTest
+{
+    private final static String VLCF_JNDI_NAME = "java:comp/env/vlj/Asrc500";
+    
+    /**
+     * Populate JNDI with a mock VistaLinkConnectionFactory that will return a
+     * MockVistaLinkConnection.
+     */
+    private static void populateJndiWithMockVlcf()
+            throws NamingException, ResourceException
+    {
+        SimpleNamingContextBuilder builder =
+                SimpleNamingContextBuilder.emptyActivatedContextBuilder();
+        final VistaLinkConnectionFactory vlcf = mock(VistaLinkConnectionFactory.class);
+        when(vlcf.getConnection(isNotNull(VistaLinkConnectionSpec.class)))
+            .thenReturn(new MockVistaLinkConnection());
+        builder.bind(VLCF_JNDI_NAME, vlcf);
+    }
+
+    @Before
+    public void setUp() throws Exception
+    {
+        populateJndiWithMockVlcf();
```

Comment:
Originally the static method was used in multiple places, but I decided to leave it separate since having its own Javadoc is useful.